src/components/TablePagination.tsx

```diff
@@ -0,0 +1,186 @@
+import {
+  Button,
+  Icon,
+  Input,
+  Select,
+  usePagination,
+} from "@canonical/react-components";
+import React, {
+  ChangeEvent,
+  Children,
+  FC,
+  HTMLAttributes,
+  PropsWithChildren,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import classnames from "classnames";
+
+const figureSmallScreen = (descriptionRef: RefObject<HTMLDivElement>) => {
+  const descriptionElement = descriptionRef.current;
+  if (!descriptionElement) {
+    return true;
+  }
+  return descriptionElement.getBoundingClientRect().width < 230;
+};
+
+const renderChildren = (
+  children: ReactNode,
+  dataForwardProp: string,
+  data: unknown[],
+) => {
+  return Children.map(children, (child) => {
+    return cloneElement(child as ReactElement, {
+      [dataForwardProp]: data,
+    });
+  });
+};
+
+const DEFAULT_PAGE_LIMITS = [50, 100, 200];
+const generatePagingOptions = (pageLimits: number[]) => {
+  return pageLimits.map((limit) => ({ value: limit, label: `${limit}/page` }));
+};
+
+interface Props {
+  data: unknown[];
+  itemName?: string;
+  containerClass?: string;
+  description?: ReactNode;
+  position?: "below" | "above";
+  dataForwardProp?: string;
+  pageLimits?: number[];
+}
+
+const TablePagination: FC<
+  PropsWithChildren<Props> & HTMLAttributes<HTMLDivElement>
+> = ({
+  data,
+  containerClass,
```

Comment:
The thinking there was to make it clear that the styling would be applicable for the container element only, but this could be solved by writing good documentation, so will revert to `className` to avoid styling conflicts as you mentioned